test/e2e/traceflow_test.go

```diff
@@ -0,0 +1,138 @@
+)
+
+func TestTraceflow(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	nodeName := workerNodeName(1)
+
+	podName0 := randName("test-pod-")
+	t.Logf("Creating Pod '%s' on '%s'", podName0, nodeName)
+	if err := data.createBusyboxPodOnNode(podName0, nodeName); err != nil {
+		t.Fatalf("Error when creating Pod '%s': %v", podName0, err)
+	}
+	defer deletePodWrapper(t, data, podName0)
+	if _, err := data.podWaitForIP(defaultTimeout, podName0, testNamespace); err != nil {
+		t.Errorf("Error when waiting for IP for Pod '%s': %v", podName0, err)
+	}
+
+	podName1 := randName("test-pod-")
+	t.Logf("Creating Pod '%s' on '%s'", podName1, nodeName)
+	if err := data.createBusyboxPodOnNode(podName1, nodeName); err != nil {
+		t.Fatalf("Error when creating Pod '%s': %v", podName1, err)
+	}
+	defer deletePodWrapper(t, data, podName1)
+	if _, err := data.podWaitForIP(defaultTimeout, podName1, testNamespace); err != nil {
+		t.Errorf("Error when waiting for IP for Pod '%s': %v", podName1, err)
+	}
+
+	tf := &v1alpha1.Traceflow{
+		Spec: v1alpha1.TraceflowSpec{
+			Source: v1alpha1.Source{
+				Namespace: testNamespace,
+				Pod:       podName0,
+			},
+			Destination: v1alpha1.Destination{
+				Namespace: testNamespace,
+				Pod:       podName1,
+			},
+			Packet: v1alpha1.Packet{
+				IPHeader: v1alpha1.IPHeader{
+					Protocol: 6,
+				},
+				TransportHeader: v1alpha1.TransportHeader{
+					TCP: &v1alpha1.TCPHeader{
+						DstPort: 80,
+					},
+				},
+			},
+		},
+	}
+
+	localConfig := rest.CopyConfig(data.kubeConfig)
+	client, err := versioned.NewForConfig(localConfig)
+	require.NoError(t, err)
+
+	var allowAllIngress *networkingv1.NetworkPolicy
+	if allowAllIngress, err = data.createNetworkPolicyAllowAllIngress("test-networkpolicy-allow-all-ingress"); err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(allowAllIngress); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+
+	tfName := randName("test-traceflow-")
+	tf.Name = tfName
+	client.OpsV1alpha1().Traceflows().Create(context.TODO(), tf, v1.CreateOptions{})
+	defer func() {
+		if err = client.OpsV1alpha1().Traceflows().Delete(context.TODO(), tfName, v1.DeleteOptions{}); err != nil {
+			t.Fatalf("Error when deleting traceflow: %v", err)
+		}
+	}()
+
+	time.Sleep(10 * time.Second)
+
+	if tf, err = client.OpsV1alpha1().Traceflows().Get(context.TODO(), tfName, v1.GetOptions{}); err != nil || tf.Status.Phase != v1alpha1.Succeeded || len(tf.Status.Results) != 1 {
+		t.Errorf("Traceflow failed: %v", err)
+	}
+
+	var obs []v1alpha1.Observation
+	if obs = tf.Status.Results[0].Observations; len(obs) != 3 {
+		t.Errorf("Traceflow number of Observations is wrong")
+	}
+	if obs[0].Action != v1alpha1.Forwarded || obs[0].Component != v1alpha1.SpoofGuard {
```

Comment:
Better to define a slice for the expected observations.

Reply:
Updated.